File: leetcode/c++/lc380.cpp
#include <vector>

using namespace std;

class RandomizedSet {
public:
    vector<int> set;

    RandomizedSet() {
        set = {};
    }
    
    bool insert(int val) {
        set.push_back(val);
    }
    
    bool remove(int val) {
        set.
    }
    
    int getRandom() {
        
    }
};

int main() {
    RandomizedSet* obj = new RandomizedSet();
    bool param_1 = obj->insert(12);
    bool param_1 = obj->insert(15);
    bool param_2 = obj->remove(12);
    int param_3 = obj->getRandom();
    return 0;
}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * RandomizedSet* obj = new RandomizedSet();
 * bool param_1 = obj->insert(val);
 * bool param_2 = obj->remove(val);
 * int param_3 = obj->getRandom();
 */

File: leetcode/python/lc380.py
import random


class RandomizedSet(object):
    def __init__(self):
        self.r_set = []
        self.size = 0

    def insert(self, val):
        """
        :type val: int
        :rtype: bool
        """
        if not val in self.r_set:
            self.r_set.append(val)
            self.size += 1
            return True
        return False

    def remove(self, val):
        """
        :type val: int
        :rtype: bool
        """
        if val in self.r_set:
            self.r_set.remove(val)
            self.size -= 1
            return True
        return False

    def getRandom(self):
        """
        :rtype: int
        """
        i = random.randint(0, self.size - 1)
        return self.r_set[i]


# Your RandomizedSet object will be instantiated and called as such:
obj = RandomizedSet()
print(obj.insert(1))
print(obj.remove(2))
print(obj.insert(2))
print(obj.getRandom())
print(obj.remove(1))
print(obj.insert(2))
print(obj.getRandom())

# [true, false, true, 2, true, false, 2]
